llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -535,8 +534,8 @@ createScalarIVSteps(VPlan &Plan, InductionDescriptor::InductionKind Kind,
   }
 
   // Truncate base induction if needed.
-  VPTypeAnalysis TypeInfo(Plan.getCanonicalIV()->getScalarType(),
-                          SE.getContext());
+  Type *CanonicalIVType = CanonicalIV->getScalarType();
+  VPTypeAnalysis TypeInfo(CanonicalIVType, CanonicalIVType->getContext());
```

Comment:
Post-commit nit, while we're here: is a context other than that of CanonicalIVType relevant, or can TypeInfo's constructor derive its second parameter from its first.

Reply:
https://github.com/llvm/llvm-project/pull/108540